src/ast_utils.h

```diff
@@ -18,6 +18,7 @@
 #define wasm_ast_utils_h
 
 #include "wasm.h"
+#include "wasm-traversal.h"
```

Comment:
Why not call it `wasm-visitor.h` after the class it declares? (or `wasm_visitor.h`? Do we have a naming convention around that? Or wasm-walker, since that's here too.). Also it might make sense to move the non-templated base class method implementations into a c++ file, especially if there's going to be several other files that include this.